toString(FunctionSchema) shows overload name

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23695 [wip] Fixed size arrays
* #23668 [wip] Call aten ops through c10 dispatcher
* #23888 Allow TensorMethods.h to include Dispatcher.h (alternative)
* #23667 Register ATen ops with c10
* #23748 Disambiguate tensor and string ops
* **#23694 toString(FunctionSchema) shows overload name**

-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
toString(FunctionSchema) shows overload name

-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

dzhulgakov commented:
@zdevito - do we use it in any user reporting? are there concerns in including overload?

zdevito commented:
We do use it in error reporting quite a bit. I haven't had time to look at this yet.

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

zdevito reviewed (approved):
This will show up in error messages, but I think it that is acceptable:

1. we really should print error messages using Python syntax for the schema anyway, meaning that they should be separate paths
2. I don't think people will get confused if the line being referenced in eq and then they see eq.Scalar in the error message. Overload names are semantic, afterall.

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)

smessmer committed:
Update on "toString(FunctionSchema) shows overload name"


-

Differential Revision: [D16611686](https://our.internmc.facebook.com/intern/diff/D16611686/)
